### PR TITLE
Cleanup/relative time

### PR DESCRIPTION
**Pull request by Leptopoda**

### CODE REVIEW

File: packages/neon/neon/lib/src/utils/relative_time.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    // Sometimes something can be messed up...
+    if (diff.isNegative) {
+      if (diff.inMinutes >= 1) {
+        text.write('-');
+      }
+      diff = Duration(microseconds: -diff.inMicroseconds);
+    }
+
+    if (diff.inMinutes < 1) {
+      text.write('now');
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/utils/relative_time.dart:22]: We should add localizations for this and the abbreviations below

File: packages/neon/neon/lib/src/widgets/relative_time.dart [hunk 2]
```diff
@@ -10,40 +13,33 @@ class RelativeTime extends StatelessWidget {
...
+  @override
+  State<RelativeTime> createState() => _RelativeTimeState();
+}
+
+class _RelativeTimeState extends State<RelativeTime> {
+  late final Timer timer;
+
+  @override
+  void initState() {
+    timer = Timer.periodic(
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/widgets/relative_time.dart:25]: I like the stream version more. Is there a particular reason you chose to use the timer?

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/neon/neon/lib/src/utils/relative_time.dart (hunk 1 above)
Comment (Leptopoda) [reply packages/neon/neon/lib/src/utils/relative_time.dart:22]: Sadly after nearly a decade intl still doesn't support duration formatting :(
IIRC there are requests dating back to 2015.

But in this case we should probably use some package. A qick search found `duration` wich does support some languages https://github.com/desktop-dart/duration/tree/master/lib/src/locale. Should we rather go this route or build our own?

File: packages/neon/neon/lib/src/widgets/relative_time.dart (hunk 2 above)
Comment (Leptopoda) [reply packages/neon/neon/lib/src/widgets/relative_time.dart:25]: There isn't any benefit.

File: packages/neon/neon/lib/src/utils/relative_time.dart (hunk 1 above)
Comment (provokateurin) [reply packages/neon/neon/lib/src/utils/relative_time.dart:22]: If you find the quality of the package is good then go for it

Comment (provokateurin) [reply packages/neon/neon/lib/src/utils/relative_time.dart:22]: The duration package looks good!

**Comment by provokateurin** [comment]
@Leptopoda can we move forward with this pr?

**Comment by Leptopoda** [comment]
The mentioned lib doesn't have a nice interface to select the locale.

I don't plan to move this PR forward any time soon. 
(Either you pick it up, we make it a draft or just close it)

**Comment by provokateurin** [comment]
Then close it I'd say

**Pull request closed without merging**